### PR TITLE
s390x: Fixing device.Bus assignment

### PR DESCRIPTION
**Pull request by zvonkok**
Fixes: #7381

The device.Bus was reset if a specific combination of configuration parameters were not met. With the new PCIe topology this should not happen anymore

### CODE REVIEW

**Comment by zvonkok** [comment]
Removing `HotplugVFIOonRootBus` which is obsolete with the latest PCI topology changes, users can set `cold_plug_vfio` or `hot_plug_vfio` either in the `configuration.toml` or via annotations.

**Comment by zvonkok** [comment]
Since this mostly affects s390x @hbrueckner @BbolroC PTAL

File: src/runtime/pkg/device/manager/manager.go [hunk 1]
```diff
@@ -195,7 +195,6 @@ func (dm *deviceManager) AttachDevice(ctx context.Context, id string, dr api.Dev
 	if !ok {
 		return ErrDeviceNotExist
 	}
-
```
Comment (BbolroC) [inline src/runtime/pkg/device/manager/manager.go:198]: Do we need to add this change to the commit? Thanks.

Comment (zvonkok) [reply src/runtime/pkg/device/manager/manager.go:198]: Of course not. Committed but not pushed.

Comment (zvonkok) [reply src/runtime/pkg/device/manager/manager.go:198]: Waiting for the static-checks to complete.

File: src/runtime/virtcontainers/sandbox.go
Comment (BbolroC) [inline src/runtime/virtcontainers/sandbox.go:1]: How do you think if we would have unit tests for the updated logic?

**Review by BbolroC — APPROVED** [review]
I've left a small knit, but overall lgtm. Thanks @zvonkok ! I appreciate your hard work.

**Comment by BbolroC** [comment]
Test results for s390x (`vfio-ap`): PASS

**Comment by zvonkok** [comment]
Depends on: https://github.com/kata-containers/tests/pull/5727

**Review by GabyCT — APPROVED** [review]

**Pull request merged**